src/mongodb-migrations.coffee

```diff
@@ -97,6 +97,7 @@ class Migrator
     allDone = (err) =>
       Promise.all(migrationsCollectionUpdatePromises).then =>
         done err, @_result
+      return
```

Comment:
I'm tracing this back to figure out why this works: http://www.quickmeme.com/img/a6/a69d84379a04cc6f56a58c1c9b8d0826ddd24c9e52644ac7a068705a7be1a6bb.jpg

Reply:
1. If this return is not put here, `allDone` will implicitly return the promise.
2. This gets returned by `runOne`, on line 104 when all migrations are done.
3. `runOne` is called on line 159 when a migration is finished, and it's return value is implicitly returned by the anonymous function
4. In the test `custom-log.coffee`, the migration's `up` function implicitly returns the result of the callback, which is the anonymous function from step 3.
5. That return value is now assigned to `donePromise` on line 150. This value is the Promise from step 1, which resolves when all promises are completed.
6. `donePromise` resolves and calls `runOne` again, which immediately calls `allDone()` again, which brings us back here where `done` is called a second time, causing the error.

Reply:
The promise doesn't actually get used anywhere. It is returned up to `@_runWhenReady` on line 178 where it is ignored, because callbacks are used at that point...

Reply:
If you can find a way to summarize that in one or two lines of comments it would probably be handy to stick it above the return.